source/testing-your-html/index.html.md.erb

```diff
@@ -0,0 +1,50 @@
+---
+title: Test your HTML matches GOV.UK Frontend
+weight: 61
+---
+
+# Test your HTML matches GOV.UK Frontend
+
+You can use our test fixtures to check you're outputting the same HTML that GOV.UK Frontend uses, when you:
+
+- create your own component macros (also called ‘partials’ or ‘templates’)
+- update GOV.UK Frontend
+
+## Using the HTML test files
+
+If you [installed GOV.UK Frontend with Node.js package manager (npm)](/installing-with-npm), you can find the `fixtures.json` file for each component in the `node_modules/govuk-frontend/govuk/components/COMPONENT-NAME/` folder, where `COMPONENT-NAME` is the name of the component.
+
+<%= warning_text('You should not test your HTML using the YAML files in the <code>src/</code> folder.') %>
+
+For example, you can find the `fixtures.json` file for the button component in the `node_modules/govuk-frontend/govuk/components/button/` folder:
+
+```js
+[
+  {
+    "name": "default",
+    "options": {
+      "text": "Save and continue"
+    },
+    "html": "<button class=\"govuk-button\" data-module=\"govuk-button\">\n  Save and continue\n</button>"
+  },
+  {
+    "name": "secondary button",
+    "options": {
+      "text": "Find address",
+      "classes": "govuk-button--secondary"
+    },
+    "html": "<button class=\"govuk-button govuk-button--secondary\" data-module=\"govuk-button\">\nFind address\n</button>"
+  },
+  ...
+]
+```
+
+Each object in the list is a different example of the component, where:
+
+- `name` is the name of the example
+- `options` are the options that generate this example's `html`
+- `html` is the HTML that GOV.UK Frontend outputs with these options
+
+For each example, pass `options` into your own macro and check the generated HTML matches `html`.
```

Comment:
I really liked the way we explained in pseudo code in the [proposal](https://github.com/alphagov/govuk-frontend/issues/1830#issuecomment-665075842) what we mean by checking for match, eg.

> For example, in pseudo-code:

```
fixtures = load_json_file("govuk/components/button/fixtures.json")
for fixture in fixtures
	expected_ouput = fixture.html
	actual_output = render_my_button(fixture.options)
	expect_equal(actual_output, expected_output)
end
```

Could we consider doing something similar here? I think this would really help to clarify on a high level what you need to do.

Reply:
Hi Hanna - I discussed this briefly with Ollie and Vanita and we think the content works ok without it. If we did add code, I wonder whether a specific real example rather than pseudocode might work better. But I think it's worth seeing if the content works without it for now, as the written description should cover everything in the code example.

Reply:
> Looks good 👍 Just left one comment.
> 
> We could also consider adding something the package README to document the HTML fixtures since it already [explains](https://github.com/alphagov/govuk-frontend/blob/master/package/govuk/components/accordion/README.md#component-options) what the macro options file is for. I can create another issue for that if it's something we want to do?

Hi @hannalaakso - ah thanks for spotting this. Are those individual READMEs in each component folder auto-generated?

Reply:
@m-green They're not autogenerated from a template but copied over from `src/govuk/components` as part of the release so each component README in `src` would need to be updated separately.

Reply:
Ah ok, thanks. I wonder whether the HTML fixtures might be slightly outside the scope of these READMEs, as these READMEs seem more about 'basic' usage. Interested to hear what others think though.

Reply:
Agree – I think the section in the component READMEs is explaining that components have options and that you can see what those options are in the Design System, rather than anything to do with the `macro-options.json` file?